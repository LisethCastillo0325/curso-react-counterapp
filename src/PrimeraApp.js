//import React, { Fragment } from 'react';
import React from 'react';
import PropTypes from 'prop-types';

//FC
const PrimeraApp = ({ saludo, subtitlo }) => {

    return (
        <>
            <h1>{ saludo }</h1>
            { /*<pre> { JSON.stringify(saludo, null, 3)} </pre> */}
            <p>{ subtitlo }</p>
        </>
    );
}

PrimeraApp.propTypes = {
    saludo: PropTypes.string.isRequired
}

PrimeraApp.defaultProps = {
    subtitlo: "Soy un subtitulo"
}

export default PrimeraApp; 