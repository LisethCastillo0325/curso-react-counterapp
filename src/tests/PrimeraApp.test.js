//import { render } from "@testing-library/react";
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import PrimeraApp from "../PrimeraApp";

describe('Pruebas en <PrimeraApp />', () => {

    // test('debe mostrar el mensaje "Hola, soy Liseth"', () => {
    //     const saludo = "Hola, soy Liseth";
    //     const wrapper = render( <PrimeraApp saludo={ saludo } /> );
    //     expect( wrapper.getByText(saludo) ).toBeInTheDocument();
    // })
    
    test('debe de mostrat <PrimeraApp/>', () => {
        const saludo = "Hola, soy Liseth";
        const wrapper = shallow( <PrimeraApp saludo = { saludo } /> );
        
        expect( wrapper ).toMatchSnapshot();

    });


    test('debe mostrar el subtitulo enviado por props', () => {
        const saludo = 'Hola, soy Liseth';
        const subTitulo = 'Las pruebas';
        const wrapper = shallow( 
            <PrimeraApp
                saludo={saludo} 
                subtitlo={subTitulo} 
            /> 
        );

        const textoParrafo = wrapper.find('p').text();

        expect(textoParrafo).toBe(subTitulo);
    });
    
    

});