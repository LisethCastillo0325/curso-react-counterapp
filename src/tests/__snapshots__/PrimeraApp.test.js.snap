// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <PrimeraApp /> debe de mostrat <PrimeraApp/> 1`] = `
<Fragment>
  <h1>
    Hola, soy Liseth
  </h1>
  <p>
    Soy un subtitulo
  </p>
</Fragment>
`;
